RelayTestAppCPP/Source/RelayTestAppCPP/GameRelayCallback.cpp

```diff
@@ -0,0 +1,77 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#include "GameRelayCallback.h"
+#include "BCBlueprintCallProxyBase.h"
+#include "ServiceName.h"
+#include "ServiceOperation.h"
+#include "Widgets/GameWidget.h"
+
+
+GameRelayCallback::GameRelayCallback(UBrainCloudWrapper *in_wrapper, IServerCallback *in_callback, class ARelayNetworkInterface *in_interface)
+{
+	Wrapper = in_wrapper;
+	Callback = in_callback;
+	Interface = in_interface;
+}
+
+void GameRelayCallback::serverCallback(ServiceName serviceName, ServiceOperation serviceOperation, const FString& jsonData)
+{
+	TSharedRef<TJsonReader<TCHAR>> reader = TJsonReaderFactory<TCHAR>::Create(jsonData);
+	TSharedPtr<FJsonObject> jsonPacket = MakeShareable(new FJsonObject());
+	bool bResponse = FJsonSerializer::Deserialize(reader, jsonPacket);
+	if(!bResponse)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("Server callback failed to deserialize jsonData"));
+		delete this;
+		return;
+	}
+
+	if (serviceName == ServiceName::AuthenticateV2 && serviceOperation == ServiceOperation::Authenticate)
+	{
+		TSharedPtr<FJsonObject> data = jsonPacket->GetObjectField(TEXT("data"));
+		FString profileId = data->GetStringField(TEXT("profileId"));
+
+		if (!profileId.IsEmpty())
+		{
+			Interface->LocalProfileID = profileId;
+		}
+		Interface->AuthenticateCallback();
+		delete this;
+	}
+	else if(serviceName == ServiceName::RTTRegistration)
+	{
+		Interface->FindLobby();
```

Comment:
I couldn't figure out how I was suppose to handle this other than deleting it within the serverError() function. The reason why I don't delete here is because when the user disconnects then I would get an error `EXCEPTION_ACCESS_VIOLATION` for BrainCloudRTTComms.cpp -> 450. From what I understood I need the lobby callback to be alive until the user disconnects otherwise the whole engine crashes with that error code.